### PR TITLE
Few fixes

### PR DESCRIPTION
**Pull request by kywch**
Hello,

* I added few entries gitignore
* When compile is False, policy is not sent to cuda, so fixed it
* Few tweaks in clean pufferl

Thanks!

### CODE REVIEW

File: pokemonred_puffer/train.py [hunk 1]
```diff
@@ -117,9 +117,9 @@ def agent_creator(env: gym.Env, args: pufferlib.namespace):
         else:
             policy = pufferlib.frameworks.cleanrl.Policy(policy)
 
+        policy = policy.to(args.train.device, non_blocking=True)
         if args.train.device == "cuda" and args.train.compile:
             torch.set_float32_matmul_precision(args.train.float32_matmul_precision)
```
Comment (thatguy11325) [inline pokemonred_puffer/train.py]:
```suggestion
        if args.train.device == "cuda":
            torch.set_float32_matmul_precision(args.train.float32_matmul_precision)
            policy = policy.to(args.train.device, non_blocking=True)
            if args.train.compile:
              # indent compile related code here
```

I know `mps` and `tpu/xla` devices exist. But lets assume the only alternative device for training is CUDA. Generally, I haven't found a reason to have compile on without CUDA so I think it makes sense to place it under the if statement.

File: pokemonred_puffer/cleanrl_puffer.py [hunk 2]
```diff
@@ -727,6 +727,7 @@ def __enter__(self):
         return self
 
     def __exit__(self, *args):
-        print("Done training. Saving data...")
+        print("Done training. Saving checkpoint...")
+        self.save_checkpoint()
```
Comment (thatguy11325) [inline pokemonred_puffer/cleanrl_puffer.py]: This was removed on purpose and I'd rather it be under a flag. Currently, we are not running evaluations and checkpoint saving has caused more headaches than solutions. It frequently errors if the wrong flags are set creating confusion for the person running this script.

**Review by thatguy11325 — CHANGES_REQUESTED** [review]
Mostly good. Left a couple of quick comments

File: pokemonred_puffer/train.py (hunk 1 above)
Comment (kywch) [reply pokemonred_puffer/train.py]: Changed as suggested.

File: pokemonred_puffer/cleanrl_puffer.py (hunk 2 above)
Comment (kywch) [reply pokemonred_puffer/cleanrl_puffer.py]: Added save_checkpoint flag to the config, and save_checkpoint() acts accordingly. Fixed the statement as well.

**Review by thatguy11325 — APPROVED** [review]

**Pull request merged**